uuid-ballerina/uuid.bal

```diff
@@ -0,0 +1,335 @@
+# ```
+#
+# + name - name
+# + namespace - namespace
+#
+# + return - UUID of type 5 as a string
+public isolated function createType5AsString(NamespaceUUID namespace, string name) returns string|Error {
+    string trimmedName = name.trim();
+    if (trimmedName.length() == 0) {
+        return UuidError("Name cannot be empty");
+    }
+    byte[] namespaceBytes = getBytesFromUUID(namespace);
+    byte[] nameBytes = trimmedName.toBytes();
+    namespaceBytes.push(...nameBytes);
+
+    byte[] uuid = crypto:hashSha1(namespaceBytes);
+
+    uuid[6] = uuid[6] & 0x0f;
+    uuid[6] = <byte>(uuid[6] | 0x50);
+    uuid[8] = uuid[8] & 0x3f;
+    uuid[8] = <byte>(uuid[8] | 0x80);
+    return getUUIDFromBytes(uuid);
+}
+
+# Returns a UUID of type 5 as a UUID record.
+# ```ballerina
+# Uuid|uuid:Error uuid5 = uuid:createType5AsRecord(uuid:NAME_SPACE_DNS, “ballerina.io”);
+# ```
+#
+# + name - name
+# + namespace - namespace
+#
+# + return - UUID of type 5 as a UUID record or error
+public isolated function createType5AsRecord(NamespaceUUID namespace, string name) returns Uuid|Error {
+    string|Error uuid = createType5AsString(namespace, name);
+    if (uuid is string) {
+        return check toRecord(uuid);
+    } else {
+        return UuidError("Failed to create uuid of type 3", uuid);
+    }
+}
+
+# Returns a nil UUID as a string.
+# ```ballerina
+# string nilUUID = uuid:nilAsString();
+# ```
+#
+# + return - nil UUID
+public isolated function nilAsString() returns string {
+    return NIL_UUID;
+}
+
+# Returns a nil UUID as a UUID record.
+# ```ballerina
+# Uuid nilUUID = uuid:nilAsRecord();
+# ```
+#
+# + return - nil UUID
+public isolated function nilAsRecord() returns Uuid {
+    Uuid nilUUID = {
+        timeLow: 0,
+        timeMid: 0,
+        timeHiAndVersion: 0,
+        clockSeqHiAndReserved: 0,
+        clockSeqLo: 0,
+        node: 0
+    };
+    return nilUUID;
+}
+
+# Test a string to see if it is a valid UUID.
+# ```ballerina
+# boolean valid = uuid:validate("4397465e-35f9-11eb-adc1-0242ac120002");
+# ```
+#
+# + uuid - UUID to be tested
+#
+# + return - true if a valied UUID, false if not
+public isolated function validate(string uuid) returns boolean {
+    return stringutils:matches(uuid, "[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}");
+}
+
+# Detect RFC version of a UUID. Returns an error if the uuid is invalid.
+# ```ballerina
+# uuid:Version|uuid:Error v = uuid:getVersion("4397465e-35f9-11eb-adc1-0242ac120002");
+# ```
+#
+# + uuid - UUID
+#
+# + return - uuid version, or error
+public isolated function getVersion(string uuid) returns Version|Error {
+    int v = getUUIDVersion(uuid);
+    if (v == 1) {
+        return V1;
+    } else if (v == 3) {
+        return V3;
+    } else if (v == 4) {
+        return V4;
+    } else if (v == 5) {
+        return V5;
+    } else {
+        if (!validate(uuid)) {
+            return UuidError("Invalid uuid string provided");
+        } else {
```

Comment:
How about using match instead of if-else?

Reply:
Updated in c690054